pkg/config/model/viper.go

```diff
@@ -610,6 +614,7 @@ func (c *safeConfig) SetConfigFile(in string) {
 	c.Lock()
 	defer c.Unlock()
 	c.Viper.SetConfigFile(in)
+	c.configSources[SourceFile].SetConfigFile(in)
```

Comment:
nit: not for this PR but it seems that there is pattern about calling the same method on the `Viper` instance and the `c.configSources[SourceFile]` on the the different methods on this file. Would we benefit from extracting that into a helper method to avoid getting into a similar situation?

Reply:
From my understanding the Viper wrapper is the place to do this double call logic, but if you find some way to improve this I'll be happy to implement this!

For example, in `SetDefault` we set the main Viper instance and the `SourceDefault` Viper instance
In `SetWithoutSource` we set the main Viper instance and the `SourceUnknown` Viper instance
etc